app/src/main/java/woowacourse/shopping/feature/order/OrderViewModel.kt

```
@@ -0,0 +1,96 @@
+    private fun updateTotalAmount() {
+        val coupon = _selectedCoupon.value
+        val discount = coupon?.calculateDiscount(cartItems, _originalAmount.value ?: 0) ?: 0
+        _totalAmount.value = (_originalAmount.value ?: 0) - discount
+    }
+
+    fun onPayClicked() {
+        viewModelScope.launch {
+            try {
+                orderRepository.addOrder(_cartItems.map { it.id })
+                _uiEvent.setValue(OrderUiEvent.OrderSuccess)
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)